### PR TITLE
Ticket441 protopaco

### PR DESCRIPTION
**Pull request by Protopaco**
### What issue is this solving?
Creates the first draft of the View / Edit Property page.  Pulls property, and tenant information from database and displays it according to the specifications.  The property information can be edited.  Property Managers & Tenants can't be edited yet.  I tried to use as many existing components as possible, and maintain existing styles.  I tried and failed to rebase the branch, so I apologize for the amount of commits. 

### Any helpful knowledge/context for the reviewer?
Is a `npm install` necessary?
Any special requirements to test?

### Please make sure you've attempted to meet the following coding standards
- [x] Code builds successfully
- [x] Code has been tested and does not produce errors
- [x] Code is readable and formatted
- [x] There isn't any unnecessary commented-out code
- [x] There aren't any unnecessary commits or files changed (shouldn't touch the /stashed folder unless the issue requests it)

If you're having trouble meeting this criteria, feel free to reach out on Slack for help!

### CODE REVIEW

**Comment by Protopaco** [comment]
I updated my earlier work, finishing the bulk of the requirements from the FIGMA.  Property Information can be viewed & edited.  Property Managers can be viewed, added & removed.  Tenants can be viewed & removed.  Code works, and there are no excessive console.logs.

Still needs pop up alert when removing Tenants.

Full functionality is waiting on some backend updates.  

Currently properties/:id .put requests don't update the tenantIDs array.  

Also, properties/:id .put doesn't recognize an empty array of property managers.  So it is possible to add/remove property managers, but not to remove all property managers.

/tenants/:id .get doesn't return lease information, so there is no unit information to plug into tenant table.

**Review by Chris-Boe — CHANGES_REQUESTED** [review]
For all new files that contain JSX, let's use the .jsx extension on the files

File: src/views/Property/PropertyView.js [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+      });
+  };
+
+
+  const removePropertyManager = (id) => {
+    property.propertyManagerIDs = []
+    for (let manager of property.propertyManager) {
+      if (manager.id !== id) property.propertyManagerIDs.push(manager.id);
+    }
+    console.log(property)
```
Comment (Chris-Boe) [inline src/views/Property/PropertyView.js]: We can toss this console log

File: src/views/Property/PropertyView.js [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+      city: values.propertyCity,
+      state: values.propertyState,
+      zipcode: values.propertyZipcode,
+      unit: values.propertyUnits,
+    };
+    updateProperty(newValues);
+  };
+
+  const updateProperty = (payload) => {
+    console.log(payload)
```
Comment (Chris-Boe) [inline src/views/Property/PropertyView.js]: Let's take out this console log

File: src/views/Property/PropertyView.js [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+
+      const property = propertyResponse.data;
+
+      setProperty(property)
+      return property;
+    }
+
+    const getTenants = async (property) => {
+
+      const tenantResponses = await Promise.all(property.tenantIDs.map(tenantID => axios.get(`${process.env.REACT_APP_PROXY}/api/tenants/${tenantID}`,
```
Comment (Chris-Boe) [inline src/views/Property/PropertyView.js]: So since we're not returning the tenantID's on this request, the page breaks

**Review by Chris-Boe — DISMISSED** [review]

**Comment by Chris-Boe** [comment]
This page's functionality will be blocked by https://github.com/codeforpdx/dwellingly-app/issues/490 and/or https://github.com/codeforpdx/dwellingly-app/issues/505 so we can get the tenants' information for a property

**Comment by Protopaco** [comment]
This commit:

- uses the .jsx extension for appropriate files
- updates functionality with updated lease information coming from property endpoint
- adds confirmation modal

To-do:

- There is an ongoing discussion about how to handle tenant lease information.  The "remove tenant" button will have to wait for the outcome to that discussion.
- Backend won't allow .put request unless there is a change in property name, receive "Property already exists error"
- minor styling tweaks

File: src/components/ManagerSearchPanel/ManagerSearchPanel.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+          });
+
+          updatedManagerOptions = updatedManagerOptions.filter(manager => !assignedPropertyManagers.includes(manager.key));
+
+          setManagerOptions(updatedManagerOptions)
+          setManagerSelection(updatedManagerOptions);
+        })
+        .catch((error) => {
+          alert(error);
+          console.log(error);
```
Comment (DaveTrost) [inline src/components/ManagerSearchPanel/ManagerSearchPanel.js]: Can you change this to a Toast?

**Review by DaveTrost — CHANGES_REQUESTED** [review]

File: src/components/PropertyManagerCard/PropertyManagerCard.scss [hunk 5]
```diff
@@ -0,0 +1,10 @@
+@import "../../scss/variables.scss";
+.property-manager-box{
+      width: 50%;
```
Comment (DaveTrost) [inline src/components/PropertyManagerCard/PropertyManagerCard.scss:3]: indentation??

File: src/components/PropertyManagerCard/PropertyManagerCard.scss [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+    height: 120px;
+    display: flex;
+    justify-content: space-between;
```
Comment (DaveTrost) [inline src/components/PropertyManagerCard/PropertyManagerCard.scss:6]: inconsistent indentation?

File: src/views/Manager/index.js [hunk 7]
```diff
@@ -32,7 +31,7 @@ const validationSchema = Yup.object().shape({
 const getManager = (context, managerId, storeInState) => {
   axios
     .get(`${process.env.REACT_APP_PROXY}/api/user/${managerId}`,
-    { Authorization: `Bearer ${context.user.accessJwt}` }
+      { Authorization: `Bearer ${context.user.accessJwt}` }
```
Comment (DaveTrost) [inline src/views/Manager/index.js]: Authorization object needs to be associated with a header key:
`{ headers: { 'Authorization': `Bearer ${context.user.accessJwt}` } }`
For a DRY way to do this, check out some of the files that use the `makeAuthHeaders` helper function

File: src/views/Manager/index.js [hunk 8]
```diff
@@ -82,27 +81,27 @@ const Manager = () => {
...
-      setEditingStatus(false);
-      Toast("Save successful!");
-    })
-    .catch((error) => {
+  const updateManager = (payload) => {
+    axios
+      .patch(`${process.env.REACT_APP_PROXY}/api/user/${id}`,
+        payload,
+        { Authorization: `Bearer ${userContext.user.accessJwt}` }
+      )
```
Comment (DaveTrost) [inline src/views/Manager/index.js:93]: Authorization object needs to be associated with a header key:
`{ headers: { 'Authorization': `Bearer ${context.user.accessJwt}` } }`
For a DRY way to do this, check out some of the files that use the `makeAuthHeaders` helper function

File: src/views/Property/PropertyView.jsx [hunk 9]
```diff
@@ -0,0 +1,399 @@
...
+    //       { headers: { Authorization: `Bearer ${userContext.user.accessJwt}` } })
+    //     .then(() => {
+    //       setEditingStatus(false);
+    //       Toast("Tenant removed");
+    //     })
+    //     .catch((error) => {
+    //       Toast(error.message);
+    //     });
+    // }
+    getProperty()
```
Comment (DaveTrost) [inline src/views/Property/PropertyView.jsx:206]: Clean up commented code?

**Review by DaveTrost — CHANGES_REQUESTED** [review]

File: src/views/Manager/index.js (hunk 7 above)
Comment (Protopaco) [reply src/views/Manager/index.js]: I'm a BIG fan of this.  I didn't see it before.  Thank you!

**Review by DaveTrost — APPROVED** [review]
Looking good!

File: src/views/Property/PropertyView.js (hunk 1 above)
Comment (DaveTrost) [reply src/views/Property/PropertyView.js]: It's gone

File: src/views/Property/PropertyView.js (hunk 3 above)
Comment (DaveTrost) [reply src/views/Property/PropertyView.js]: Not sure when this changed, but the id's are present now

File: src/components/ManagerSearchPanel/ManagerSearchPanel.js (hunk 4 above)
Comment (DaveTrost) [reply src/components/ManagerSearchPanel/ManagerSearchPanel.js]: Thanks for the change @Protopaco !

**Review by Chris-Boe — APPROVED** [review]
Nice work! I noticed I was getting a 400 error on the backend when editing a property details without changing the name, but I see @aedwardg has a fix in a PR for that already.

**Pull request merged**